compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/inference/FirBuilderInferenceSession.kt

```diff
@@ -216,3 +226,26 @@ class ConeComposedSubstitutor(val left: ConeSubstitutor, val right: ConeSubstitu
         return left.substituteOrNull(rightSubstitution ?: type)
     }
 }
+
+class FirStubTypeTransformer(
+    override val session: FirSession,
+    private val substitutor: ConeSubstitutor
+) : FirAbstractTreeTransformer<Void?>(phase = FirResolvePhase.IMPLICIT_TYPES_BODY_RESOLVE) {
+
+    override fun transformTypeRef(typeRef: FirTypeRef, data: Void?): CompositeTransformResult<FirTypeRef> {
```

Comment:
This is not needed since you override `transformResolvedTypeRef`

Reply:
Right, I somehow started with this and forgot to revert it, thanks.